### PR TITLE
Ineligible(nuclear) detonation sites, blacklists nuke from being activated at LZs, enclosed areas & areas with the area_flag "CANNOT_NUKE"

### PR DESCRIPTION
**Pull request by mrhugo13**
## About The Pull Request
Adds "CANNOT_NUKE" area flag and the relevant checks for it.
Adds "CANNOT_NUKE" flag to multiple areas with images shown later below as to where they are put
Adds a global "nuke_ineligible_sites" list & adds areas with the CANNOT_NUKE flag to it on area mapload initialize alongside a bunch of other checks like enclosed areas or inside LZ
Adds TGUI to show names of the above nuke_activation_sites on the nuke UI (BIG THANKS TO GULPER/@neocloudy ON THE TGMC DISCORD) (WHICH WAS THEN FURTHER HELPED ON BY THE GREAT AND ONLY @SandPoot)
Puts LZ areas into turfs with shutters on them so they count as NEAR_FOB, fixing both *that* problem alongside making me use NEAR_FOB instead for preventing fob nukes.
Added an Enclosed Area area to orion outpost and then used it on all the rocks so that it better integrates with PR.

Fairly well tested on a local server although further TM-testing may be needed as to check wheter the currently available detonation areas are balanced as well to gauge reception & criticism.
dreamseeker_2024-06-03_15-13-06

## Why It's Good For The Game
Hey so it kinda sucks that marines can just plasmacut themselves into a cave wall 10 tiles deep and make it impossible for xenos to respond accordingly? Completely ruining the point of nuke defence? Making it so braindead easy to defend that it would require a grief-level mess-up for xenos to possibly win? Yeah this is a response to that.

With this mappers/coders can accordingly moderate this by putting the CANNOT_NUKE flag into areas they want accessible to be nuked, I have done this before posting the PR both as to not brick the nuke but also in accordance with the ideas that:
* Cannot be activated inside Enclosed Areas.
* Cannot be detonated inside FOB (though usually still outside or a bit away from it)
* Cannot be nuked in areas otherwise determined with the CANNOT_NUKE area_flags bitfield
~~Cannot be detonated in deep underground caves (like enclosed areas which the plasmacutter can cut into)~~
~~Nuke detonation areas should neither be too open nor too closed so both sides should have a decent fight.~~
~~Neither too far nor too close to the FOB so that marines can still possibly evac without any helping vehicles, although possibly risking the nuke if they leave it too unguarded.~~
~~Staying fairly decently away from caves unless where otherwise hard to find a good spot.~~
~~Generally keeping it at around 4 spots on every map so that there is still fairly plenty of options and variability.~~

This should make nuke defence alot more engaging and active for everyone involved, overall improving fun factor by about 30-50 percent, aswell as providing better tools for coders/mappers to balance this part of a round.

Also orion outpost enclosed areas added both for mappers and coders and whatever else so that there is a better divide between "here is where someone is gonna user their plasmacutter" areas and "plasmacutters begone" areas
## Changelog 
:cl: Vondiech/Citruses (with help from Neocloudy and SandPoot)
balance: The nuke can now NOT be activated in areas like the LZs, enclosed areas (the type plasmacutters cut into) and certain other areas as listed on the "Ineligible detonation sites" part of the nuclear bomb UI.
code: Added CANNOT_NUKE area_flag for mappers/coders to designate areas where the nuke timer can't be activated, also moves some areas around so NEAR_FOB now accurately always includes FOB.
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
maybe to go further with this , we should add a guaranteed detonation countdown, so that the nuke can't be stopped when its on its last 60 seconds for example , and marines can actually evacuate or get stopped by xenos and it turns into a minor.

**Comment by SimplySmiley123** [comment]
I would like to add:
Make the areas possible to figure out without looking up on github/in code for love of fucking god.

Also not being able to put it in an FOB with how it is currently means it would be guaranteed loss or marines are already winning hard enough for Tad evac to be possible unless there were to be a failsafe on low enough countdown. Mostly so if the nuke area is indeed too far, it doesn't cuck marines from being able to evacuate in time _and_ detonate the nuke.

**Comment by chizzy376** [comment]
> Also not being able to put it in an FOB with how it is currently means it would be guaranteed loss

So instead of a guaranteed win it becomes a "guaranteed" loss? What? You're really blowing this out of proportions

**Comment by Lumipharon** [comment]
this seems overly complicated and frankly kinda arbitrary.

If this was more as a blacklist (i.e. can't pc deep into walls and nuke there, or certain other bad spots) it would make sense.
Otherwise, if you want the nuke to go off in a very specific spot just... make it unmovable and you have to activate it where it is, putting it in a sensible spot to begin with.

**Comment by Lauwma** [comment]
This is a bit stupid, most areas you put as the nuke area can't have tad land nearby which means marines can evacuate. If you want this go this route make those areas LZs since all LZs are already standardized as siegeable.

**Comment by chizzy376** [comment]
> this seems overly complicated and frankly kinda arbitrary.
> 
> If this was more as a blacklist (i.e. can't pc deep into walls and nuke there, or certain other bad spots) it would make sense. Otherwise, if you want the nuke to go off in a very specific spot just... make it unmovable and you have to activate it where it is, putting it in a sensible spot to begin with.

That's not the point of the PR and is frankly not even a good choice. The point of the PR is to turn the act of nuking into another objective, as marines don't just backtrack to the fob with the nuke in hand. This still fulfills the "one last stand" between the two teams, but makes it more favorable towards xenomorphs as the marines have to fight on xeno territory.

**Comment by chizzy376** [comment]
> This is a bit stupid, most areas you put as the nuke area can't have tad land nearby which means marines can evacuate. If you want this go this route make those areas LZs since all LZs are already standardized as siegeable.

LZs are siegable but are next to unbreakable once the nuke is there, as the entire marine force has nothing to do but defend the fob. This PR will still let you nukehold, but you have to properly set up in enemy territory and have to figure out transport logistics. More teamwork, more encouragement for roles to work with each other.

**Comment by mrhugo13** [comment]
I HAVE ARISEN FROM MY SLEEPY GRAVE TO ANSWER QUESTIONS & CONCERNS

> [MLGTASTICa]
> maybe to go further with this , we should add a guaranteed detonation countdown, so that the nuke can't be stopped when its on its last 60 seconds for example , and marines can actually evacuate or get stopped by xenos and it turns into a minor.

I agree on this point although it would be out of scope for this specific PR, would definetly be interested in as a follow-up PR. Last minute haste & action as xenos either manage to catch marines stepping outside the metaphorical circle or marines expertly getting out of there alive like some sort of "cool guys don't look at explosions" action movie.

> [SimplySmiley123]
> I would like to add: Make the areas possible to figure out without looking up on github/in code for love of fucking god.
> 
> Also not being able to put it in an FOB with how it is currently means it would be guaranteed loss or marines are already winning hard enough for Tad evac to be possible unless there were to be a failsafe on low enough countdown. Mostly so if the nuke area is indeed too far, it doesn't cuck marines from being able to evacuate in time _and_ detonate the nuke.

For the first point I'd like to see some TM reception & criticism on it to see if marines genuinely don't know what the area names correspond to since i feel like "DESIGNATED DETONATION SPOT: HYDROPHONICS" when there is maybe one or two very hydrophonics themed rooms on the map should be fairly obvious. If not then i'll consider checking if I can fitz around with the area colors so it is more clear for marines (hey have we got the kinda tgui noticebox stripes thing but for the minimap? would be cool!)

Second point, the failsafe I feel would be for a follow-up PR but otherwise I disagree with making the LZ's of maps detonation sites since either A: Marines have not built up FOB and therefor the defence will just be unfun for them or B: Marines have built up FOB and pretty much have an unbelieveably strong position to defend from (if marines have not wiped completely, FOB sieges can take upward of half an hour or more, which i've experienced on multiple different occasions and have gotten some less fun memories from)

Overall, unless a TM shows a failsafe is required to make this more aptly fun for both sides i'd consider it out of scope for this PR and to be done in a follow-up PR shortly after if this were to be merged by the powers that be.

> [Lumipharon]
> this seems overly complicated and frankly kinda arbitrary.
> 
> If this was more as a blacklist (i.e. can't pc deep into walls and nuke there, or certain other bad spots) it would make sense. Otherwise, if you want the nuke to go off in a very specific spot just... make it unmovable and you have to activate it where it is, putting it in a sensible spot to begin with.

A blacklist would be significantly longer than a whitelist and way more exhausting for anyone to go through, so I feel a whitelist is more appropriate although if required to deal with balancing problems then the whitelist would easily be able to account for more places. I disagree on the unmoveable nuke part since it would likely make nuke defence very same-y (unless one were to make alot of modular maps for a single spot for each map?) while this PR would atleast give marines more varied places that mappers/coders can directly plot out on the map. Maybe explore this path later if any TM of this PR were to poke catastrophic holes in it.

>[Lauwma]
> This is a bit stupid, most areas you put as the nuke area can't have tad land nearby which means marines can evacuate. If you want this go this route make those areas LZs since all LZs are already standardized as siegeable.

I'm unsure as to which areas you are referring to since a fair amount of them have open spaces available close by for the tad to make use of, although these nuke areas can be moved if it indeed turns out the tad can't adequately evacuate any of them. Once again though i disagree on LZ's being used as detonation sites, even if they are built for sieges, since the minimum nuke timer (6 minutes iirc?) is incredibly difficult to do in time for a group of xenos if the marine force hasn't been sufficiently wiped beforehand and the FOB built to an adequate degree, many a round have gone on for way too long because of stalemates that starts and ends at a marine-populated FOB.


In general, I'd like to get some in-game feedback via TM or whatever other means so I can see what this PR would otherwise need to be more conclusively fun-inducing for rounds. Though once again some of the ideas flagged up here (nuke timer window for marines to run away in time) is something that I feel should be explored as a part of a quick follow-up PR so as to more accurately determine the specific balancing needs & receptions for it without making this PR go out of scope.

**Comment by Lauwma** [comment]
All the maps beside delta station and prison station have at least one suitable area for tad to land nearby. I'd suggest removing all the others areas that don't have an open space WITHOUT A STEEL ROOF ON TOP (tad can't land anywhere it wants now).

**Comment by mrhugo13** [comment]
> [Lauwma]
> All the maps beside delta station and prison station have at least one suitable area for tad to land nearby. I'd suggest removing all the others areas that don't have an open space WITHOUT A STEEL ROOF ON TOP (tad can't land anywhere it wants now).
> 
> Also LZs may be difficult to siege but they get broken into most of the times while marines are defending disks. It would give a reason for marines to defend LZ and for Xenos to breach it before Marines get the nuke.

I'll keep the other areas for the sake of variance however I'll make sure to add some better nuke spots on Delta station & Prison Station.

For the LZ thing, that may be something worth exploring? But I'd like to recieve additional in-game feedback & criticism beforehand because I'm still overall suspicous of allowing the use of a full FOB for nuke defence, xenos spending time cracking the FOB while marines are getting disks with little resistance would likely cause frustration as well as less action/fun for both sides, though I admit once again this may be worth exploring later just because it might touch-up the dynamic of the FOB during the round well enough for marines to spend some resources defending it. 

tl;dr yeah i'll spruce up the delta/prison spots but the LZ/FOB defence thing I'd like in-game feedback on before implementing.

**Comment by mrhugo13** [comment]
One nuke spot added to both delta station & prison station.

One here on prison station
bild

And one here on delta station
bild

**Comment by Lauwma** [comment]
Also I wonder but will it appear on the tacmap that you can only nuke in these areas ?

**Comment by mrhugo13** [comment]
>[Lauwma]
> Also I wonder but will it appear on the tacmap that you can only nuke in these areas ?

Currently only appears on the nuke UI as the name of that area (i.e. there are 4 spots on desparity that you can nuke and it shows up on the nuke UI as seen in the gif posted in the main body of the PR), if testing or otherwise calls for it might be worth looking into changing the area colors on the tac-/minimap into a specific "NUKE GOES HERE" color

**Comment by SimplySmiley123** [comment]
> > Also not being able to put it in an FOB with how it is currently means it would be guaranteed loss
> 
> So instead of a guaranteed win it becomes a "guaranteed" loss? What? You're really blowing this out of proportions

It's about being able to evacuate at all and still being able to blow up the nuke.
Too far away from an FOB and it's a guaranteed loss, Tadpole not included as it's apparently wanted gone / is to not be relied upon and can't be guaranteed to live through the op.

**Comment by mrhugo13** [comment]
Here are the designated detonation sites on riptide with a thanks to @chizzy376 for consulting on it
StrongDMM_2024-05-14_14-13-57

**Comment by mrhugo13** [comment]
> I only ask for the nuke can still able be used to blow up the ship

The marines unknowing concession is that no longer will one be able to nuke shipside through sheer incompetence (although it may remove some funnies i'm unsure if anyone would earnestly want this for any reason other than griefing)

**Comment by Lumipharon** [comment]
Doubling down on this.
As above, if the issue this is trying to address is very cheesy nuke spots (i.e. PC deep into walls etc), then this should be handled by a blacklist, and simply blacklist out of bound areas, plus any edge case cheesy areas if required.

If its the issue of moving the nuke at all to make it easier defend then it should be a larger change of making nuke unmovable and having maps position the nuke with this in mind.

**Comment by Jackal-boop** [comment]
You could just remove the ability to transport the nuke on the tadpole. Seems like a simpler fix

File: code/game/area/area.dm [hunk 1]
```diff
@@ -87,11 +87,15 @@
...
 	if(!static_lighting)
 		blend_mode = BLEND_MULTIPLY
-
 	reg_in_areas_in_z()
 
 	update_base_lighting()
 
+	if(area_flags & NUKE_AREA) // This adds the area to the list of areas displayed on the nuke detonation site UI
+		if(src in GLOB.nuke_activation_sites) // To prevent duplicate entries into the list
+			return
```
Comment (TiviPlus) [inline code/game/area/area.dm:96]:
```suggestion
		if(get_area_name(src) in GLOB.nuke_activation_sites) // To prevent duplicate entries into the list
			return
```

File: code/game/objects/machinery/nuclearbomb.dm [hunk 2]
```diff
@@ -257,7 +258,9 @@
 	if(!anchored)
 		balloon_alert(user, "anchors not set")
 		return
-
+	if(!(area.area_flags & NUKE_AREA)) // if area we're activating in does not have the area flag NUKE_AREA, return.
+		balloon_alert(user, "Warning! This thermonuclear device needs to be armed at one of the designated detonation sites.")
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:262]: this needs to be shorter

File: code/game/objects/machinery/nuclearbomb.dm [hunk 3]
```diff
@@ -257,7 +258,9 @@
 	if(!anchored)
 		balloon_alert(user, "anchors not set")
 		return
-
+	if(!(area.area_flags & NUKE_AREA)) // if area we're activating in does not have the area flag NUKE_AREA, return.
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:261]: useless comment
```suggestion
	if(!(area.area_flags & NUKE_AREA))
```

File: code/game/objects/machinery/nuclearbomb.dm [hunk 4]
```diff
@@ -249,6 +249,7 @@
 
 ///Toggles the timer on or off
 /obj/machinery/nuclearbomb/proc/toggle_timer(mob/user)
+	var/area/area = get_area(src)
 	if(exploded)
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:253]: you should declare this just before you use it

File: tgui/packages/tgui/interfaces/NuclearBomb.tsx [hunk 5]
```diff
@@ -7,11 +7,13 @@ import {
   NoticeBox,
   ProgressBar,
   Section,
+  Stack,
 } from '../components';
 import { Window } from '../layouts';
 
 type NuclearBombData = {
   status: string;
+  nuke_activation_sites: any;
```
Comment (TiviPlus) [inline tgui/packages/tgui/interfaces/NuclearBomb.tsx:16]: Why is this using any? It should use the correct type (string[] i think)

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest lgtm

**Comment by mrhugo13** [comment]
> [Lumipharon]
> Doubling down on this. As above, if the issue this is trying to address is very cheesy nuke spots (i.e. PC deep into walls etc), then this should be handled by a blacklist, and simply blacklist out of bound areas, plus any edge case cheesy areas if required.
> 
> If its the issue of moving the nuke at all to make it easier defend then it should be a larger change of making nuke unmovable and having maps position the nuke with this in mind.

I'll try to see if I can satisfingly implement the former in switching from whitelist to blacklist.


> [Jackal-Boop]
> You could just remove the ability to transport the nuke on the tadpole. Seems like a simpler fix

This PR is in regards to detonating the nuke in a not-too-cheesy-spot rather than transporting it to that spot.


> [Tiviplus]
> Rest lgtm

Will have to fitz aroound with some stuff to get it up to snuff with Lumi's demands but i'll make sure to work everything around the reviews & requested changes!

File: tgui/packages/tgui/interfaces/NuclearBomb.tsx (hunk 5 above)
Comment (mrhugo13) [reply tgui/packages/tgui/interfaces/NuclearBomb.tsx]: Done!

File: code/game/objects/machinery/nuclearbomb.dm (hunk 4 above)
Comment (mrhugo13) [reply code/game/objects/machinery/nuclearbomb.dm]: Done!

File: code/game/objects/machinery/nuclearbomb.dm (hunk 3 above)
Comment (mrhugo13) [reply code/game/objects/machinery/nuclearbomb.dm]: Done!

File: code/game/objects/machinery/nuclearbomb.dm (hunk 2 above)
Comment (mrhugo13) [reply code/game/objects/machinery/nuclearbomb.dm]: Done!

File: code/game/area/area.dm (hunk 1 above)
Comment (mrhugo13) [reply code/game/area/area.dm]: Done!

File: code/_globalvars/lists/objects.dm [hunk 6]
```diff
@@ -42,6 +42,8 @@ GLOBAL_LIST_EMPTY(nuke_disk_generators)
 GLOBAL_LIST_EMPTY(nuke_list)
 GLOBAL_LIST_EMPTY(active_nuke_list)
 GLOBAL_LIST_EMPTY(nuke_spawn_locs)
+/// List of where /obj/machiner/nuclearbomb can be activated
+GLOBAL_LIST_EMPTY(nuke_activation_sites)
```
[the remark this answers is not in the document]
Comment (mrhugo13) [reply code/_globalvars/lists/objects.dm]: Done!

**Comment by mrhugo13** [comment]
> > [Lumipharon]
> > Doubling down on this. As above, if the issue this is trying to address is very cheesy nuke spots (i.e. PC deep into walls etc), then this should be handled by a blacklist, and simply blacklist out of bound areas, plus any edge case cheesy areas if required.
> > If its the issue of moving the nuke at all to make it easier defend then it should be a larger change of making nuke unmovable and having maps position the nuke with this in mind.
> 
> I'll try to see if I can satisfingly implement the former in switching from whitelist to blacklist.
> 
> > [Jackal-Boop]
> > You could just remove the ability to transport the nuke on the tadpole. Seems like a simpler fix
> 
> This PR is in regards to detonating the nuke in a not-too-cheesy-spot rather than transporting it to that spot.
> 
> > [Tiviplus]
> > Rest lgtm
> 
> Will have to fitz aroound with some stuff to get it up to snuff with Lumi's demands but i'll make sure to work everything around the reviews & requested changes!


Now done as requested and the PR title & body changed accordingly!

**Comment by Lumipharon** [comment]
Whats the issue with nuking in fob, exactly? Its the natural place to defend.

When I said blacklist I didn't mean blacklist a huge number of areas so its basically the same result as a whitelist, I meant like, if there are SPECIFIC areas that are particularly bad for gameplay (i.e. the out of bounds area that solid mineral walls might occupy behind fob, etc), just blacklist those.

**Comment by mrhugo13** [comment]
> [Lumipharon]
> Whats the issue with nuking in fob, exactly? Its the natural place to defend.
> 
> When I said blacklist I didn't mean blacklist a huge number of areas so its basically the same result as a whitelist, I meant like, if there are SPECIFIC areas that are particularly bad for gameplay (i.e. the out of bounds area that solid mineral walls might occupy behind fob, etc), just blacklist those.

RE: Nuke in fob, I basically think doing so is on-par with plasmacutter cheesing since, _unless your force is wiped to the point where you can't cover all fronts of a FOB_, the FOB basically becomes impenetrable with the amount of firepower you can concentrate into it, like we're talking xenos not being able to make use of defiler or boiler or any other siege castes cause they simply get shot so quick they instantly die if they get near meanwhile the nuke timer has already ticked halfway down and the marines have an easy evac on their hands (additionaly, tanglefoot OB spam exists). Ultimately ends up becoming a "I stick the nuke in the corner of the FOB no xeno can penetrate and wait out the timer" which is uncomfortably close to the same logic current PC-cheese uses hence why I feel it's better to shake up the gameplay by forcing marines to employ other locations to detonate from, where a more balanced last-struggle can occur.

As for the second sentence, this does more or less blacklist those areas, PC-cut areas are defined by being "deep underground" (that and i'm quite frankly doing marines a favor by telling them not to waste time trying to detonate deep in caves so they can, as a large part of this PR is, have a more balanced and engaging last struggle), NEAR_FOB areas are ones primarily defined by either being FOB (which i have already explained isn't fun to try siegeing in 6 minutes or lose) or being so near it that it's basically an extension of the FOB.

For a conclusion/tl;dr: FOB or NEAR_FOB areas aren't good places to have a fun/balanced 6 minute nuke struggle because, unless marines have lost too many and have yet to replenish, it will just lead to a curbstomp with no real counter.

I will say though that, if it turns out via TM or otherwise that marines really honestly-to-god can't win unless they get to put their nuke inside their mega-death-fort, i'm willing to look into stripping the NEAR_FOB check.

**Comment by SimplySmiley123** [comment]
IMO if this were to be a thing it would need areas designed around it rather then picking whatever fancies you on the map.
But then this would just end up like an LZ FOB, drop pod /tadpole there or march all the way from LZ to it, setup a FOB in the predetermined nuke-able area, go around doing objectives and come back to it. It would end up the same way anyway.

Hell, if anything this just adds to the reasons why you should never make an FOB in the LZ, cuz besides the nuke and the off-chance that Tadpole died (and if that does happen marines will just go for the minor right where they got the nuke anyway just to end the round faster as by the point marines get to the nuke xenos won't put up much of a fight and will often just delay the round instead), there's not much reason to make anything in the LZ.

**Comment by mrhugo13** [comment]
> > [Lumipharon]
> > Whats the issue with nuking in fob, exactly? Its the natural place to defend.
> > When I said blacklist I didn't mean blacklist a huge number of areas so its basically the same result as a whitelist, I meant like, if there are SPECIFIC areas that are particularly bad for gameplay (i.e. the out of bounds area that solid mineral walls might occupy behind fob, etc), just blacklist those.
> 
> RE: Nuke in fob, I basically think doing so is on-par with plasmacutter cheesing since, _unless your force is wiped to the point where you can't cover all fronts of a FOB_, the FOB basically becomes impenetrable with the amount of firepower you can concentrate into it, like we're talking xenos not being able to make use of defiler or boiler or any other siege castes cause they simply get shot so quick they instantly die if they get near meanwhile the nuke timer has already ticked halfway down and the marines have an easy evac on their hands (additionaly, tanglefoot OB spam exists). Ultimately ends up becoming a "I stick the nuke in the corner of the FOB no xeno can penetrate and wait out the timer" which is uncomfortably close to the same logic current PC-cheese uses hence why I feel it's better to shake up the gameplay by forcing marines to employ other locations to detonate from, where a more balanced last-struggle can occur.
> 
> As for the second sentence, this does more or less blacklist those areas, PC-cut areas are defined by being "deep underground" (that and i'm quite frankly doing marines a favor by telling them not to waste time trying to detonate deep in caves so they can, as a large part of this PR is, have a more balanced and engaging last struggle), NEAR_FOB areas are ones primarily defined by either being FOB (which i have already explained isn't fun to try siegeing in 6 minutes or lose) or being so near it that it's basically an extension of the FOB.
> 
> For a conclusion/tl;dr: FOB or NEAR_FOB areas aren't good places to have a fun/balanced 6 minute nuke struggle because, unless marines have lost too many and have yet to replenish, it will just lead to a curbstomp with no real counter.
> 
> I will say though that, if it turns out via TM or otherwise that marines really honestly-to-god can't win unless they get to put their nuke inside their mega-death-fort, i'm willing to look into stripping the NEAR_FOB check.

alright i got like the perfect screenshot of an example of this on round id 27159 (i will admit xenos were eating dirt but this would still be impossible to crack in 6 minutes when compared to SD which is about on par with 20 minutes)
dreamseeker_2024-05-27_16-59-38

**Comment by SandPoot** [comment]
[JoUNInqMHU](https://github.com/mrhugo13/TerraGov-Marine-Corps/pull/2)
UI suggestion

**Comment by blackdav123** [comment]
listing 30 ineligible areas is not useful

**Comment by Lumipharon** [comment]
why are the blacklists still so long? 
image

why are caves even blacklisted? You're blacklisting deep underground which is most of caves, when there is zero reason for this? Again I am really confused by how arbitrary these lists are.

**Comment by CoastalCookie** [comment]
I feel like the inability to put a nuke on specific spots is a cool idea but why _blacklist more open locations where marines are good?_ That seems more xeno biased then actual gameplay effectiveness... Blocking deep caves so plasma cutting is a good call... 
But in the end if xenos let marines easily push the nuke to FOB its slightly their fault as they didn't really do much to force the marines to set nuke there on the spot or damaged FOB to the point of being useless, and tad evac they can also just have shrike screech the entire marine force out a moment before it leaves or just flat out screech most of the force out of tad or even disable it when possible. Its not like the marines currently are professionals who all group up on one tile with t60's. They need those good defend points and removing them may majorly impact the game. Like the concept though of forcing original holding locations but this seems more forcing a defeat from a difficult fight then anything else

**Comment by mrhugo13** [comment]
Deep underground caves (including enclosed areas) -> Enclosed areas

so now the current blacklist is:
FOB
Near FOB
Enclosed areas (areas you plasmacut into)

**Comment by mrhugo13** [comment]
> [SandPoot]
> [JoUNInqMHU](https://github.com/mrhugo13/TerraGov-Marine-Corps/pull/2) [ JoUNInqMHU ](https://github.com/mrhugo13/TerraGov-Marine-Corps/pull/2) [ ](https://github.com/mrhugo13/TerraGov-Marine-Corps/pull/2) UI suggestion

Didn't realize this was a link to a PR on my own fork for this PR, will merge with it!

File: code/game/objects/machinery/nuclearbomb.dm [hunk 7]
```diff
@@ -257,7 +258,10 @@
 	if(!anchored)
 		balloon_alert(user, "anchors not set")
 		return
-
+	var/area/area = get_area(src)
+	if(get_area_name(area) in GLOB.nuke_ineligible_site)
+		balloon_alert(user, "ineligible detonation site.")
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:263]:
```suggestion
	if(get_area_name(area) in GLOB.nuke_ineligible_site)
		balloon_alert(user, "ineligible detonation site")
```

**Review by TiviPlus — APPROVED** [review]
